src/mmw/apps/geoprocessing_api/validation.py

```diff
@@ -69,3 +71,104 @@ def validate_uuid(uuid):
         return False
 
     return uuid == str(u)
+
+
+def validate_gwlfe_prepare(data):
+    area_of_interest = data.get('area_of_interest')
+    wkaoi = data.get('wkaoi')
+    huc = data.get('huc')
+    layer_overrides = data.get('layer_overrides', {})
+
+    if not check_gwlfe_only_one([area_of_interest, wkaoi, huc]):
+        error = ('Invalid parameter: One and only one type of area object'
+                 ' (area_of_interest, wkaoi, or huc) is allowed')
+        raise ValidationError(error)
+
+    not_valid_layers = check_layer_overrides_keys(layer_overrides)
+
+    if not_valid_layers:
+        error = create_layer_overrides_keys_not_valid_msg(not_valid_layers)
+        raise ValidationError(error)
+
+    if ('__LAND__' in layer_overrides.keys() and
+            not check_land_layer_overrides(layer_overrides)):
+        error = create_layer_overrides_land_not_valid_msg(LAND_LAYER_OVERRIDES)
+        raise ValidationError(error)
+
+    if ('__STREAMS__' in layer_overrides.keys() is not None and
+            not check_streams_layer_overrides(layer_overrides)):
+        error = 'Invalid __STREAMS__ param: Must be `drb`, `nhdhr`, or `nhd`'
+        raise ValidationError(error)
+
+
+def validate_gwlfe_run(input, job_uuid):
+    if not check_gwlfe_only_one([input, job_uuid]):
+        error = ('Invalid parameter: Only one type of prepared input'
+                 '(input JSON or job_uuid) is allowed')
+        raise ValidationError(error)
+
+    if not check_gwlfe_run_input(input):
+        error = ("Invalid input: Please use the full result "
+                 "of gwlf-e/prepare endpoint's result object")
```

Comment:
Let's make a follow up issue to call out specific keys that are missing from the payload.

Reply:
Made #3509 for this.